compiler/rustc_codegen_llvm/src/lib.rs

```diff
@@ -177,9 +177,34 @@ impl WriteBackendMethods for LlvmCodegenBackend {
     type ThinData = back::lto::ThinData;
     type ThinBuffer = back::lto::ThinBuffer;
     fn print_pass_timings(&self) {
-        unsafe {
-            llvm::LLVMRustPrintPassTimings();
-        }
+        let msg = unsafe {
+            let mut size = 0;
+            let cstr = llvm::LLVMRustPrintPassTimings(&mut size as *mut usize);
+            if cstr.is_null() {
+                "failed to get pass timings".into()
+            } else {
+                let timings = std::slice::from_raw_parts(cstr as *const u8, size);
+                let timings = std::str::from_utf8(timings).unwrap();
```

Comment:
This creates a slice that refers to the data from `cstr`. The next line frees this data, making the Rust slice dangling.

Reply:
Also don't even need utf8 validation, you could just write the bytes directly to stdout:

```rust
std::io::stdout().write_all(timings).unwrap();
```

this way you don't have the issue of preserving the data beyond the `free`